chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -239,21 +241,19 @@ trait TransformerMacros extends MappingMacros with TargetConstructorMacros with
       To: Type
   ): Either[Seq[TransformerDerivationError], Tree] = {
 
-    From.valueClassMember
-      .map { member =>
-        Right {
-          mkTransformerBodyTree0(config.derivationTarget) {
-            q"$srcPrefixTree.${member.name}"
-          }
-        }
-      }
-      .getOrElse {
-        // $COVERAGE-OFF$
-        Left {
-          Seq(CantFindValueClassMember(From.typeSymbol.name.toString, To.typeSymbol.name.toString))
-        }
-        // $COVERAGE-ON$
-      }
+    val fromValueClassMember = From.valueClassMember.toRight(
+      Seq(CantFindValueClassMember(From.typeSymbol.name.toString, To.typeSymbol.name.toString))
+    )
+
+    for {
+      fromValueClassMember <- fromValueClassMember
+      fromValueClassMemberType = fromValueClassMember.returnType
+      fromMemberAccessTree = q"$srcPrefixTree.${fromValueClassMember.name}"
+      transformerBodyTree <- resolveRecursiveTransformerBody(fromMemberAccessTree, config)(
+        fromValueClassMemberType,
+        To
+      )
+    } yield mkTransformerBodyTree0(transformerBodyTree.target)(transformerBodyTree.tree)
```

Comment:
You should use `config.derivationTarget` - this is the most probable reason why the tests are failing.
Same applies to the remaining rules.